orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -398,10 +428,30 @@ def _create_training_session(self):
         # 0:Verbose, 1:Info, 2:Warning. 3:Error, 4:Fatal. Default is 2.
         session_options.log_severity_level = int(self._verbosity)
 
-        self._training_session = onnxruntime.training.TrainingAgent(self._onnx_training.SerializeToString(),
+        return session_options, providers, provider_options
+
+    def _create_training_session(self, session_options, providers, provider_options):
+        self._training_session = onnxruntime.training.ExecutionAgent(self._onnx_training.SerializeToString(),
                                                                     session_options, providers, provider_options)
 
-    def _build_training_graph(self, *inputs, **kwargs):
+    def _create_inference_session(self, session_options, providers, provider_options):
+        self._onnx_inference = onnx.load_model_from_string(self._module_gradient_graph_builder.get_inference_model())
+        self._inference_session = onnxruntime.training.ExecutionAgent(
+            self._onnx_inference.SerializeToString(),
+            session_options, providers, provider_options, False)
```

Comment:
Style nit: Could we model the two "modes" of `ExecutionAgent` (distinguished by the last argument) as two different classes (e.g. `InferenceAgent` and `TrainingAgent` which both extend `ExecutionAgent`... although the inheritance might be unnecessary since it's Python!)?